### PR TITLE
[pleasereview] #17 (conversion to wei) & re-export/use U256/Address instead of type conversion helpers

### PR DESCRIPTION
**Pull request by snd**
- [x] add one test for every function with values verified by https://github.com/ethereum/wiki/wiki/JavaScript-API#web3towei
- [x] add documentation

### CODE REVIEW

File: solaris/src/wei.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    pub static ref KETHER: U256 = U256::from(10).pow(21.into());
+    pub static ref ETHER: U256 = U256::from(10).pow(18.into());
+    pub static ref FINNEY: U256 = U256::from(10).pow(15.into());
+    pub static ref SZABO: U256 = U256::from(10).pow(12.into());
+    pub static ref GWEI: U256 = U256::from(10).pow(9.into());
+    pub static ref MWEI: U256 = U256::from(10).pow(6.into());
+    pub static ref KWEI: U256 = U256::from(10).pow(3.into());
+}
+
+pub fn from_tether<T: Into<U256>>(v: T) -> U256 {
```
Comment (tomusdrw) [inline solaris/src/wei.rs]: Could you also add docs while you are here? We should later add `#!deny(missing_docs)` to the whole crate.

File: solaris/src/wei.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+pub fn from_ada<T: Into<U256>>(v: T) -> U256 {
+    v.into() * *KWEI
+}
+
+pub fn from_wei<T: Into<U256>>(v: T) -> U256 {
+    v.into()
+}
+
+pub fn convert(v: [u8; 32]) -> U256 {
```
Comment (tomusdrw) [inline solaris/src/wei.rs]: Maybe call this `from_bytes32` ?

File: solaris/src/wei.rs (hunk 1 above)
Comment (snd) [reply solaris/src/wei.rs]: sure

**Comment by snd** [comment]
added tests and documentation. not the biggest fan of all the testing boilerplate and duplication but i don't see a reasonable way to cut down on that. can i get a review?

File: solaris/src/convert.rs [hunk 3]
```diff
@@ -35,3 +36,17 @@ pub fn bytes32(s: &str) -> [u8; 32] {
 pub fn uint(num: u64) -> types::U256 {
     num.into()
 }
+
+pub fn u256_from_bytes32(v: [u8; 32]) -> types::U256 {
+    v.into()
+}
+
+pub fn convert_u256(x: types::U256) -> bigint::uint::U256 {
```
Comment (tomusdrw) [inline solaris/src/convert.rs]: Those functions are only used internally, right? Maybe mark them as `pub(crate)`

File: solaris/src/convert.rs [hunk 4]
```diff
@@ -35,3 +36,17 @@ pub fn bytes32(s: &str) -> [u8; 32] {
 pub fn uint(num: u64) -> types::U256 {
     num.into()
 }
+
+pub fn u256_from_bytes32(v: [u8; 32]) -> types::U256 {
```
Comment (tomusdrw) [inline solaris/src/convert.rs]: This doesn't match convention in this module.
Either: `uint_from_bytes32()` or just `u256` or `convert::uint_from_raw/convert::raw::to_uint(x)` (the reason for the last one being that, we use `convert:raw::uint(5)` to generate bytes)

File: solaris/src/convert.rs (hunk 3 above)
Comment (snd) [reply solaris/src/convert.rs]: done

File: solaris/src/convert.rs (hunk 4 above)
Comment (tomusdrw) [reply solaris/src/convert.rs]: A bit too verbose imho: `uint_from_u64` or `uint_from_into_u256`?
I'm in favour of last proposal, where this method get's moved to `raw`. `raw` is implicitly `bytes`, so you have:
```
convert::raw::to_uint(convert:raw::uint(5)) == convert::uint(5)
```

Anyway, this module will be gone when we migrate primitives. @snd let me know if you want to work on introducing primitives to `ethabi`.

**Comment by snd** [comment]
as discussed on riot re-exported `U256` and `Address` and used their conversions instead of custom conversion functions. can i get another review?

**Pull request merged**